### PR TITLE
Feat/engagement form

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/bm-commitment`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #... 
| License          | BSD 3-Clause

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report installation#16864.0 ✘  
* Stage 1
  * Install ✘

File: packages/manager/apps/dedicated/client/app/app.js [hunk 1]
```diff
@@ -260,6 +260,7 @@ angular
       });
 
       $state.defaultErrorHandler((error) => {
+        console.log(error);
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/app.js:263]:
```suggestion
        
```

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/commitment/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const moduleName = 'ovhManagerBillingAutorenewCommitmentLazyLoading';
+
+angular.module(moduleName, ['ui.router', 'oc.lazyLoad']).config(
+  /* @ngInject */ ($stateProvider) => {
+    $stateProvider.state(
+      'app.account.billing.autorenew.service.commitment.**',
+      {
+        url: '/commitment',
+        lazyLoad: ($transition$) => {
+          const $ocLazyLoad = $transition$.injector().get('$ocLazyLoad');
```
Comment (radireddy) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/commitment/index.js:15]: 👍

File: packages/manager/modules/billing/package.json [hunk 3]
```diff
@@ -11,12 +11,21 @@
   "license": "BSD-3-Clause",
   "author": "OVH SAS",
   "main": "./src/index.js",
+  "dependencies": {
+    "@ovh-ux/manager-config": "^2.0.0",
```
Comment (radireddy) [inline packages/manager/modules/billing/package.json:15]:
```suggestion
    "@ovh-ux/manager-config": "^2.0.0",
```
could these be added in peerDependencies?

File: packages/manager/modules/billing/src/components/commitment/Commitment.class.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      },
+    }).format();
+  }
+
+  get commitmentType() {
+    return this.configuration.type;
+  }
+
+  isPeriodic() {
+    return this.commitmentType === 'periodic';
```
Comment (radireddy) [inline packages/manager/modules/billing/src/components/commitment/Commitment.class.js:63]: 'periodic' and 'upfront' can be defined in constants file

File: packages/manager/modules/billing/package.json (hunk 3 above)
Comment (marie-j) [reply packages/manager/modules/billing/package.json:15]: @radireddy it's a non angular dependency

File: packages/manager/modules/billing/src/components/commitment/commitment.controller.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    if (this.engagement) {
+      [this.model.engagement] = commitments;
+    }
+
+    const upfront = commitments.find((commitment) => commitment.isUpfront());
+    const periodic = commitments.find((commitment) => commitment.isPeriodic());
+
+    if (upfront && periodic) {
+      this.discount = Math.floor(
+        (periodic.totalPrice.value / upfront.totalPrice.value - 1) * 100,
```
Comment (frenautvh) [inline packages/manager/modules/billing/src/components/commitment/commitment.controller.js:105]: if you want to round down a negative number use Math.ceil ;)
```
x = (41 / 51 - 1) * 100 = -25.45
Math.floor(x) = -26
```

**Comment by ovh-cds** [comment]
CDS Report build#16956.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#16972.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#16972.1 ✘  
* 
  * Build ✘

File: packages/manager/modules/billing/src/components/commitment/commitment.controller.js (hunk 5 above)
Comment (marie-j) [reply packages/manager/modules/billing/src/components/commitment/commitment.controller.js:105]: @frenautvh thanks for the tip but if the number is negative then we have an API issue :)

**Comment by ovh-cds** [comment]
CDS Report test#17006.0 ✘  
* Stage 1
  * lint ✔ 
  * SemVer ✘

**Comment by ovh-cds** [comment]
CDS Report build#17006.0 ✘  
* 
  * Build ✘

File: packages/manager/modules/billing/src/components/commitment/Commitment.class.js (hunk 4 above)
Comment (marie-j) [reply packages/manager/modules/billing/src/components/commitment/Commitment.class.js]: They could but isn't it redundant as we created the class to abstract those values and use those values just through the `isUpfront` and `isPeriodic` functions

File: packages/manager/modules/billing/src/components/commitment/commitment.html [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                        }"
+                            ></span>
+                        </oui-select-picker-description>
+                        <oui-select-picker-footer>
+                            <span
+                                class="oui-heading_6"
+                                data-translate="billing_commitment_price_month"
+                                data-translate-values="{
+                                    price: duration.monthlyPrice
+                                    }"
```
Comment (JeremyDec) [inline packages/manager/modules/billing/src/components/commitment/commitment.html:82]:
```suggestion
                                }"
```

**Review by JeremyDec — APPROVED** [review]

File: packages/manager/modules/billing/src/components/commitment/commitment.controller.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    $q,
+    $translate,
+    $window,
+    atInternet,
+    BillingService,
+    BillingCommitmentService,
+    ovhPaymentMethod,
+    RedirectionService,
+  ) {
+    this.$http = $http;
```
Comment (antleblanc) [inline packages/manager/modules/billing/src/components/commitment/commitment.controller.js:17]: Not used I guess.

```suggestion
```

**Pull request merged**